macro-pdfviewer-webjar/src/main/webjar/pdfviewer-customizations.js

```diff
@@ -50,8 +50,8 @@
   }
   // PDFVIEWER-13: Allow author of the macro to delegate its view right on the PDF document.
   function setViewRights() {
-    var viewerOnly = window.location.search.indexOf('PDFViewerService') !== -1;
-    if (viewerOnly) {
+    var viewerOnly = window.frameElement.getAttribute("data-delegated");
```

Comment:
Does the iframe always have the same origin as the host page (e.g. even when viewing a PDF file specified using an external URL)? Because if not, then the value of ``window.frameElement`` will be ``null``. See https://developer.mozilla.org/en-US/docs/Web/API/Window/frameElement#value

If the iframe always has the same origin, then it's simpler to use:

    var viewerOnly = window.frameElement.dataset.delegated;